### PR TITLE
Set `Flatrate Conditions` to `NULL` for both `Purchase` and `Sales Order`

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/19069

### CODE REVIEW

File: backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java [hunk 1]
```diff
@@ -134,7 +137,7 @@ public void onPriceListChangeInterceptor(final I_C_Order order)
 		onPriceListChange(order);
 	}
 
-	@CalloutMethod(columnNames = I_C_Order.COLUMNNAME_M_PriceList_ID)
+	@CalloutMethod(columnNames = I_C_Order.COLUMNNAME_C_DocTypeTarget_ID)
```
Comment (metas-rc) [inline backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java:140]: why did you change this?

Comment (riadmen) [reply backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java]: just an error, corrected

File: backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java [hunk 2]
```diff
@@ -506,6 +509,29 @@ public void updateDescriptionFromDocType(final I_C_Order order)
...
+	}, ifColumnsChanged = {
+			I_C_Order.COLUMNNAME_C_DocTypeTarget_ID
+	})
+	public void removeFlatRateConditionsForCallOrder(final I_C_Order order)
+	{
+		if (order.getC_DocTypeTarget_ID() <= 0)
+		{
+			return;
+		}
+		final I_C_DocType dt = Services.get(IDocTypeDAO.class).getById(order.getC_DocTypeTarget_ID());
```
Comment (metas-rc) [inline backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java:524]: please, move this at the beginning of the class with the rest of the services.

Comment (riadmen) [reply backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java]: done

**Review by metas-rc — APPROVED** [review]

**Review by metas-rc — APPROVED** [review]
